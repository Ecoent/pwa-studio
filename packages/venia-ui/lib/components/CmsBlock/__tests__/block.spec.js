import React from 'react';
import { createTestInstance } from '@magento/peregrine';
import Block from '../block';

jest.mock('../../../classify');

test('renders a Block component', () => {
    const blockProps = {
        content: ''
    };
    const component = createTestInstance(<Block {...blockProps} />);

    expect(component.toJSON()).toMatchSnapshot();
});

test('renders a Block component with all props configured and Page Builder rich content', () => {
    const blockProps = {
        content:
            '<div data-content-type="row" data-appearance="contained" data-element="main"><div data-enable-parallax="0" data-parallax-speed="0.5" data-background-images="{}" data-element="inner" style="justify-content: flex-start; display: flex; flex-direction: column; background-position: left top; background-size: cover; background-repeat: no-repeat; background-attachment: scroll; border-style: solid; border-color: rgb(255, 0, 0); border-width: 5px; border-radius: 2px; margin: 0px 0px 10px; padding: 10px;"></div></div>'
    };
    const component = createTestInstance(<Block {...blockProps} />);

    expect(component.toJSON()).toMatchSnapshot();
});

test('renders a Block component with HTML content', () => {
    const blockProps = {
        content: '<h1>Testing HTML content</h1>'
    };
    const component = createTestInstance(<Block {...blockProps} />);

    expect(component.toJSON()).toMatchSnapshot();
});
