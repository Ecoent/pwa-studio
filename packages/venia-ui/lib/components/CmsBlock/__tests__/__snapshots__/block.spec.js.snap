// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a Block component 1`] = `
<div
  dangerouslySetInnerHTML={
    Object {
      "__html": "",
    }
  }
/>
`;

exports[`renders a Block component with HTML content 1`] = `
<div
  dangerouslySetInnerHTML={
    Object {
      "__html": "<h1>Testing HTML content</h1>",
    }
  }
/>
`;

exports[`renders a Block component with all props configured and Page Builder rich content 1`] = `
<div
  className="root contained"
  style={
    Object {
      "backgroundAttachment": "scroll",
      "backgroundColor": null,
      "backgroundImage": null,
      "backgroundPosition": "left top",
      "backgroundRepeat": "no-repeat",
      "backgroundSize": "cover",
      "border": "solid",
      "borderColor": "rgb(255, 0, 0)",
      "borderRadius": "2px",
      "borderWidth": "5px",
      "display": "flex",
      "flexDirection": "column",
      "justifyContent": "flex-start",
      "marginBottom": "10px",
      "marginLeft": "0px",
      "marginRight": "0px",
      "marginTop": "0px",
      "minHeight": null,
      "paddingBottom": "10px",
      "paddingLeft": "10px",
      "paddingRight": "10px",
      "paddingTop": "10px",
      "textAlign": "",
    }
  }
/>
`;
